Fix/update compatibility matrix 1.14.x

yrsuthari opened the pull request:
## Related issue #1547

<!--
Please link the GitHub issue this pull request resolves in the format of `#1234`. If you discussed this change
with a maintainer, please mention her/him using the `@` syntax (e.g. `@JimBugwadia`).

If this change neither resolves an existing issue nor has sign-off from one of the maintainers, there is a
chance substantial changes will be requested or that the changes will be rejected.

You can discuss changes with maintainers in the [Kyverno Slack Channel](https://kubernetes.slack.com/).
-->

## Proposed Changes

<!--
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. 

***NOTE***: If this PR results in new or altered behavior which is user facing, you **MUST** read and follow the steps outlined in the [PR documentation guide](pr_documentation.md) and add Proof Manifests as defined below.
-->

This PR updates the compatibility matrix in the installation documentation to include Kyverno version 1.14.x and its corresponding Kubernetes compatibility range (1.29 - 1.32).

## Checklist

<!--
Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
-->

- [x] I have read the [contributing guidelines](https://github.com/kyverno/website/blob/main/CONTRIBUTING.md).
- [x] I have inspected the website preview for accuracy.
- [x] I have signed off my issue.

JimBugwadia commented on content/en/docs/installation/_index.md:
```diff
@@ -46,6 +46,7 @@ Kyverno follows the same support policy as the Kubernetes project (N-2 policy) i
 | 1.11.x                         | 1.25           | 1.28           |
```
```suggestion
```

JimBugwadia commented:
@yrsuthari - can you please check the CI errors?

JimBugwadia reviewed (changes requested):
@yrsuthari - looks like there are number of unrelated changes included in the PR. Can you please create a "clean" PR with only the version matrix change?

yrsuthari commented:
> @yrsuthari - looks like there are number of unrelated changes included in the PR. Can you please create a "clean" PR with only the version matrix change?

@JimBugwadia Thank you for your review. I've addressed your concerns by cleaning up the PR, but I wanted to explain why there are still some additional files included:
1. The main change is to content/en/docs/installation/_index.md for the compatibility matrix update
2. The other included files are necessary for the PR to pass CI:
  - .github/workflows/check-links.yaml: Added rate limiting handling (--accept "429") to fix the link checker failures
  - The modified files in content/en/docs/kyverno-cli/ and content/en/policies/other/: These are auto-generated files updated by running make codegen and must be included to pass the verify-codegen test

While these additional changes may seem unrelated, they're required for the PR to pass all CI checks. I've kept the changes minimal and focused on only what's needed to successfully update the compatibility matrix.
